### PR TITLE
Move to new travis system

### PR DESCRIPTION
**Pull request by jmesterh**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,71 @@
...
+    - stage: deploy-snapshots
+      script:
+        - echo "deploy-snapshots"
+        # When push to master occurs, all the versions should be -SNAPSHOTs and we will auto-deploy
+        - mvn --batch-mode -e -DskipTests=true deploy
+    - stage: release-check
+      script:
+        - echo "release-check"
+        # Ensure we don't have any external SNAPSHOT dependencies
+        - mvn --batch-mode release:prepare -DskipTests=true -DdryRun=true
```
Comment (ctjoreilly) [inline .travis.yml]: Change this line to:
mvn --batch-mode release:prepare -DskipTests -Darguments=-DskipTests -DdryRun=true
you need the extra '-Darguments=...' to work correctly with the release plugin.

**Comment by jmesterh** [comment]
Still need to configure travis with tokens for github and sonatype.

File: pom.xml [hunk 2]
```diff
@@ -252,6 +242,13 @@
   </distributionManagement>
 
   <build>
+    <extensions>
+      <extension>
+        <groupId>io.packagecloud.maven.wagon</groupId>
+        <artifactId>maven-packagecloud-wagon</artifactId>
+        <version>0.0.6</version>
+      </extension>
+    </extensions>
```
Comment (garricko) [inline pom.xml]: Might not need this, since we no longer use packagecloud

File: pom.xml [hunk 3]
```diff
@@ -18,10 +18,15 @@
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>jsr305</artifactId>
+      <version>3.0.2</version>
+    </dependency>
```
Comment (garricko) [inline pom.xml]: Does this need to be a dependency users of the library will inherit? Or is it just a build time need?

Comment (jmesterh) [reply pom.xml]: Not needed at all, not sure how that got there.

**Pull request merged**